cpp/src/io/parquet/page_enc.cu

```diff
@@ -13,15 +13,15 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+#include <io/parquet/parquet_gpu.hpp>
 #include <io/utilities/block_utils.cuh>
-#include "parquet_gpu.h"
 
 namespace cudf {
 namespace io {
 namespace parquet {
 namespace gpu {
 // Spark doesn't support RLE encoding for BOOLEANs
-#define ENABLE_BOOL_RLE 0
+constexpr bool enable_bool_rle = false;
```

Comment:
Why change this from a preprocessor definition to a constexpr? I understand with a #define we can override the value at compile time to enable the code paths, but this way aren't all the code paths that check for `enable_bool_rle` now permanently dead code?

Reply:
I get your point, but prefer `constexpr` here because the disabled code at least needs to compile, while with `ifdef` it can completely break without us catching it. 
How about this?
```
#ifdef ENABLE_BOOL_RLE
constexpr bool enable_bool_rle = true;
#else
constexpr bool enable_bool_rle = false;
#endif
```

Reply:
The other option to explore could be to convert the code paths into template functions over the boolean flag.
```

template <bool is_bool_rle_enabled>
__device__
void
encode(page_enc_state_s *const s, uint32_t rle_numvals, int32_t dict_bits, bool flush, uint32_t dtype, uint32_t t)
{
  if (dtype == BOOLEAN) {
    PlainBoolEncode(s, rle_numvals, flush, t);
  } else
  {
    RleEncode(s, rle_numvals, dict_bits, flush, t);
  }
}

template <>
__device__
void
encode<true>(page_enc_state_s *const s, uint32_t rle_numvals, int32_t dict_bits, bool flush, uint32_t dtype, uint32_t t)
{
  RleEncode(s, rle_numvals, dict_bits, flush, t);
}
```
And then this section in the kernel :
```
if ((!enable_bool_rle) && (dtype == BOOLEAN)) {
  PlainBoolEncode(s, rle_numvals, (cur_row == s->page.num_rows), t);
} else {
  RleEncode(s, rle_numvals, dict_bits, (cur_row == s->page.num_rows), t);
}
```
can be converted to this :

`encode<enable_bool_rle>(s, rle_numvals, dict_bits, (cur_row == s->page.num_rows), dtype, t);`

Reply:
I prefer the current use of `enable_bool_rle`, especially with the #define way of initialization. Not a strong preference though.